src/components/molecules/AuthButton/AuthUserProfile/index.tsx

```diff
@@ -0,0 +1,33 @@
+"use client";
+
+import { getMyProfile } from "@/apis/profile";
+import Button from "@/components/atoms/Button";
+import { deleteToken } from "@/utils/user";
+import { useQuery } from "@tanstack/react-query";
+import { useRouter } from "next/navigation";
+
+interface Props {
+  setIsLogin: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+function AuthUserProfile({ setIsLogin }: Props) {
+  const { data } = useQuery(["/api/users/mine"], getMyProfile);
+  const router = useRouter();
+
+  const handleLogout = () => {
+    deleteToken();
+    setIsLogin(false);
+    router.refresh();
+  };
+
+  return (
+    <>
+      <span className="mr-2 text-sm text-gray-500">{data?.data?.response?.name}님</span>
+      <Button styleType="white" rounded="md" size="sm" onClick={handleLogout}>
+        로그아웃
+      </Button>
```

Comment:
여기는 onClickLogout 핸들러를 Props로 받도록 변경하면 됩니다!

```suggestion
interface Props {
  onClickLogout: () => void;
}

function AuthUserProfile({ onClickLogout }: Props) {
  const { data } = useQuery(["/api/users/mine"], getMyProfile);
  const router = useRouter();

  const handleLogout = () => {
    deleteToken();
    onClickLogout();
    router.refresh();
  };

  return (
    <>
      <span className="mr-2 text-sm text-gray-500">{data?.data?.response?.name}님</span>
      <Button styleType="white" rounded="md" size="sm" onClick={handleLogout}>
        로그아웃
      </Button>
```

Reply:
완료